metaflow/plugins/kubernetes/kubernetes_client.py

```diff
@@ -60,6 +64,48 @@ def get(self):
 
         return self._client
 
+    def _find_active_jobs(self, flow_name, run_id=None, user=None, field_selector=None):
+        flow_hash = hashed_label(flow_name)
+        results = self._client.BatchV1Api().list_namespaced_job(
+            namespace=self._namespace,
+            label_selector="metaflow.org/flow-hash=%s" % flow_hash,
+            field_selector=field_selector,
+        )
+        for job in results.items:
+            match = (
+                job.status.active
+                and (
+                    run_id is None
+                    or job.metadata.annotations["metaflow/run_id"] == run_id
+                )
+                and (user is None or job.metadata.annotations["metaflow/user"] == user)
+            )
+            if match:
+                yield job
+
+    def list(self, flow_name, run_id, user):
+        results = self._find_active_jobs(flow_name, run_id, user)
+
+        return list(results)
+
+    def kill_jobs(self, flow_name, run_id, user, echo):
+        jobs = self._find_active_jobs(
+            flow_name, run_id, user, field_selector="status.successful==0"
+        )
+
+        def _kill_job(job):
```

Comment:
does this work for argo-workflows?

Reply:
as-is, no this does not work. I was leveraging the existing `RunningJob` for the kill logic, but from what I understood we do not create Kubernetes Jobs for runs on Argo Workflows, as the workflow dag wraps the pods directly? As such, the lookup for executing jobs doesn't return anything to be terminated.

I would maybe leave argo workflows out of scope for this PR, as there is a more direct way to issue termination to a workflow on that side instead of individually killing pods.

If we want this to apply to argo-workflows as well then I can change the lookup to search for active pods instead of jobs, and introduce the kill logic separately.